src/main/java/org/folio/marc4ld/service/ld2marc/mapper/impl/agent/AgentMapper.java

```diff
@@ -65,24 +68,35 @@ protected char getRelationTextSubfield() {
   @Override
   public boolean test(ResourceEdge resourceEdge) {
     return resourceEdge.getPredicate() != null
-      && SUPPORTED_PREDICATES.contains(resourceEdge.getPredicate())
-      && getSupportedTypes().containsAll(resourceEdge.getTarget().getTypes());
+      && (isAgentEdge(resourceEdge) || isRelationEdge(resourceEdge));
   }
 
   @Override
   public DataField apply(ResourceEdge resourceEdge) {
+    if (isRelationEdge(resourceEdge)) {
+      return EMPTY_DATA_FIELD;
+    }
     var resource = resourceEdge.getTarget();
     var dataField = marcFactory.newDataField(getTag(resourceEdge), getIndicator1(resource), SPACE);
     addRepeatableSubfields(dataField, resource);
     addNonRepeatableSubfields(dataField, resource);
-    addRelationCodes(dataField, resource);
-    addRelationNames(dataField, resource);
+    addRelationCodes(dataField, resourceEdge.getSource());
```

Comment:
Hi @askhat-abishev , The current logic will mix up the roles of all contributors into $e and $4 if a work has more than one contributor as shown in the graph below. right?

<img width="534" alt="image" src="https://github.com/user-attachments/assets/5e1e19a6-8809-4d9f-8f40-f57bcb767de0">

Reply:
fixed